pymech/dataset.py

```diff
@@ -0,0 +1,76 @@
+"""Interface for reading files as xarray datasets."""
+from pathlib import Path
+import numpy as np
+import xarray as xr
+
+from .neksuite import readnek
+
+
+def open_dataset(path):
+	"""Helper function for opening a file as an xarray dataset."""
+	path = Path(path)
+	if path.suffix.startswith('.f'):
+		cls = NekDataset
+	else:
+		raise NotImplementedError(
+			"Filetype: {} is not supported.".format(path.suffix)
+		)
+
+	return cls(path).open()
+
+
+class NekDataset:
+	"""Interface for converting Nek field files into xarray datasets"""
+	def __init__(self, path):
+		self.field = readnek(path)
+		self.axes = ("z", "y", "x")
+
+	def meshgrid_to_dim(self, mesh):
+		"""Reverse of np.meshgrid. This method extracts one-dimensional
+		coordinates from a cubical array format for every direction
+
+		"""
+		dim = np.unique(np.round(mesh, 8))
+		return dim
```

Comment:
Possible caveat: I am unsure how this approach of extracting 1D grids would work on inclined / curved geometries.